src/devices/Shtc3/Shtc3.cs

```diff
@@ -0,0 +1,290 @@
+        /// </summary>
+        public void Reset()
+        {
+            Write(Register.SHTC3_RESET);
+        }
+
+        private int _id = 0;
+
+        /// <summary>
+        /// Sensor Id
+        /// </summary>
+        public int Id
+        {
+            get
+            {
+                if (_id == 0)
+                {
+                    Id = ReadId();
+                }
+
+                return _id;
+            }
+            set
+            {
+                _id = value;
+            }
+        }
+
+        /// <summary>
+        /// Read Id
+        /// </summary>
+        private int ReadId()
+        {
+            Write(Register.SHTC3_ID);
+
+            Span<byte> readBuff = stackalloc byte[3];
+
+            _i2cDevice.Read(readBuff);
+
+            // check 8-bit crc
+            if (!CheckCrc8(readBuff.Slice(0, 2), readBuff[2]))
+            {
+                return 0;
+            }
+
+            var id = (readBuff[0] << 8) | readBuff[1];
+
+            // check the result match to the SHTC3 product code
+            if (!ValidShtc3Id(id))
+            {
+                return 0;
+            }
+
+            return id;
+        }
+
+        /// <summary>
+        /// Check the match to the SHTC3 product code
+        /// Table 15  while bits 11 and 5:0 contain the SHTC3 specific product code 0b_0000_1000_0000_0111
+        /// </summary>
+        /// <param name="id">Id to test</param>
+        /// <returns></returns>
+        private static bool ValidShtc3Id(int id)
+        {
+            return (id & 0b_0000_1000_0011_1111) == 0b_0000_1000_0000_0111;
+        }
+
+        /// <summary>
+        /// 8-bit CRC Checksum Calculation
+        /// </summary>
+        /// <param name="data">Raw Data</param>
+        /// <param name="crc8">Raw CRC8</param>
+        /// <returns>Checksum is true or false</returns>
+        private static bool CheckCrc8(ReadOnlySpan<byte> data, byte crc8)
+        {
+            // Details in the Datasheet table 16 P9
+            byte crc = CRC_INIT;
+            for (int i = 0; i < 2; i++)
+            {
+                crc ^= data[i];
+
+                for (int j = 8; j > 0; j--)
+                {
+                    if ((crc & 0x80) != 0)
+                    {
+                        crc = (byte)((crc << 1) ^ CRC_POLYNOMIAL);
+                    }
+                    else
+                    {
+                        crc = (byte)(crc << 1);
+                    }
+                }
+            }
+
+            return crc == crc8;
+        }
+
+        private void Write(Register register)
+        {
+            byte msb = (byte)((short)register >> 8);
+            byte lsb = (byte)((short)register & 0xFF);
+
+            Span<byte> writeBuff = stackalloc byte[]
+            {
+                msb,
+                lsb
+            };
+
+            _i2cDevice.Write(writeBuff);
+
+            // wait SCL free
+            DelayHelper.DelayMilliseconds(20, false);
```

Comment:
please point to doc page where is 20 coming from

Reply:
just based on SHT3x, maybe the table 5 page 4  suggest to reduce to 15 ms ?

Reply:
fine as is